File: simple.rb
# frozen_string_literal: true

word_count = Hash.new(0)

$stdin.each_line do |line|
  line.downcase.split.each do |word|
    word_count[word] += 1
  end
end

word_count.sort_by { |_key, value| value }
          .each { |pair| puts "#{pair[0]}: #{pair[1]}" }

File: simple.jl
function wc(io, counter=Dict{String,Int}())
    for line in eachline(io), word in split(line)
        lword = lowercase(word)
        counter[lword] = get(counter, lword, 0) + 1
    end
    counter
end

sortdict(counter) = sort!(collect(counter), by=x->x[2], rev=true)

wc(stdin) |> sortdict .|> kv->println(kv[1], " ", kv[2]) 
